fuzz/Cargo.toml

```diff
@@ -10,7 +10,7 @@ publish = false
 cargo-fuzz = true
 
 [dependencies]
-honggfuzz = { version = "0.5.55", default-features = false }
+honggfuzz = { version = "0.5.56", default-features = false }
```

Comment:
I don't think this is needed The lock files should be enough.

Reply:
Yes, but I think it won't cause any damage and is less a footgun if we bump it up.

Reply:
It could if someone needs to downgrade for some reason (e.g. to investigate problems).

Reply:
Yes, but then someone would change the `Cargo.toml`.
I think it can cause way more damage if we just pin versions in the lockfile.
The real question is: "Are we going to need to downgrade `honggfuzz`?" From what I gather out of #2923, I don't think so.

Reply:
Why would it? Updating only lock files is literally considered standard in Rust. And for good reasons.

Reply:
I think we should bump it in the Cargo.toml file. Eventually 0.5.55 will break with some rustc or LLVM update and we won't notice. Bumping the toml also forces people to update their lockfiles, which will reduce the amount of "the version of honggfuzz installed does not match the one in your lockfile, please do something about that" annoying errors people have to deal with. (Of course, conversely, if we bump Cargo.toml then their only option is to upgrade their version of honggfuzz, which may cause this error to show up on other crates..)

But I don't feel very strongly about this.

utACK this PR, and utACK a variant that doesn't update the .toml file.

Sorry for not reviewing yet -- I need to update the copy of honggfuzz in my local CI infra to review this and I've been procrastinating on that.

Reply:
I think it would be more annoying in the future for the person who needs to update the lockfile again if the toml file isn't updated.  They would need to read the lockfile to create a toml file, then apply the update with which to build a new lockfile with which sounds annoying.  Conversely if the toml file is brought along with in the commit history, they only need to update the toml file to create the new lockfile.

Reply:
~~I think it would be more annoying in the future for the person who needs to update the lockfile again if the toml file isn't updated. They would need to read the lockfile to create a toml file, then apply the update with which to build a new lockfile with which sounds annoying. Conversely if the toml file is brought along with in the commit history, they only need to update the toml file to create the new lockfile.~~

Actually that's wrong.  I guess it doesn't matter, however I think it's easier on the eyes to read a toml file if people want to know what versions are.

Reply:
> I think it's easier on the eyes to read a toml file if people want to know what versions are.

You can't really do that since the lock file has priority. If you read the toml file you might unknowingly be using a higher version. Toml only says the lowest one. I actually never look into toml when figuring out the version I *actually use* for exactly this reason.

Reply:
That's a fair point with how semvar works I guess.  I wonder if it's possible to auto-build a toml file from a lockfile for that reason?

Reply:
Why would you do that? Just run `cargo tree` to inspect, it's not really that hard.

Reply:
It sounds like you're saying nobody should ever use a .toml file expect maybe to bootstrap the project?

Reply:
Stop strawmaning me. Toml is for specifying which dependencies with which version ranges (should) work, lock is for specifying which actually are used. Thus toml should be updated whenever the dependencies change (including minimal version).

Reply:
heh I had to lookup what stawmaning is.  I guess I don't have a preference on this so I'm fine with whatever.

Reply:
and fwiw my questions are not framed in a way to strawman anyone, I'm simply trying to ask questions to better understand.

Reply:
Guess I need to work on my delivery of how I ask questions

Reply:
Yeah, the question reminded me of that meme "so what you're saying" woman in an interview with Jordan Peterson, though it wasn't nearly as bad, so I hope it's not too offensive to mention her. :man_shrugging:

Reply:
Had to lookup that meme :joy: